rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -282,6 +282,41 @@ impl<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>> ArrowBlockfileRe
         }
     }
 
+    pub(crate) async fn get_at_index(
+        &'me self,
+        index: usize,
+    ) -> Result<(&'me str, K, V), Box<dyn ChromaError>> {
+        let mut block_offset = 0;
+        let mut block = None;
+        let sparse_index_len = self.sparse_index.len();
+        for i in 0..sparse_index_len {
+            let uuid = {
+                let sparse_index_forward = self.sparse_index.forward.lock();
+                *sparse_index_forward.iter().nth(i).unwrap().1
+            };
+            block = self.get_block(uuid).await;
+            match block {
+                Some(b) => {
+                    if block_offset + b.len() > index {
+                        break;
+                    }
+                    block_offset += b.len();
+                }
+                None => {
+                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
+                }
+            }
+        }
+        let block = block.unwrap();
+        let res = block.get_at_index::<'me, K, V>(index - block_offset);
+        match res {
```

Comment:
Appreciate if you could use more meaningful name for res.0, res.1 and res.2

Reply:
will do, thanks, good point